### PR TITLE
Inject new artsy/reaction auction routes

### PR DESCRIPTION
**Pull request by dblandin**
Working on migrating the bidder registration flow first.

Depends upon https://github.com/artsy/reaction/pull/2605

cc/ @dleve123 @erikdstock

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please don't include .js files, we want to be using TypeScript found: <a href="https://github.com/dblandin/force/blob/inject-new-bidder-registration-flow/src/desktop/assets/auction_reaction.js">auction_reaction.js</a>.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The staging version of Metaphysics has breaking changes, this might not affect your build, but could mean your queries don't work the way you expect. See below for breaking changes:</td>
    </tr>
  </tbody>
</table>


- :warning: Field 'CommerceOrder.createdAt' changed type from 'CommerceDateTime!' to 'String!'
- :warning: Field 'CommerceOrder.lastApprovedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOrder.lastSubmittedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOrder.stateExpiresAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOrder.stateUpdatedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOrder.updatedAt' changed type from 'CommerceDateTime!' to 'String!'
- :warning: Field 'CommerceBuyOrder.createdAt' changed type from 'CommerceDateTime!' to 'String!'
- :warning: Field 'CommerceBuyOrder.lastApprovedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceBuyOrder.lastSubmittedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceBuyOrder.stateExpiresAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceBuyOrder.stateUpdatedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceBuyOrder.updatedAt' changed type from 'CommerceDateTime!' to 'String!'
- :warning: Field 'CommerceOffer.createdAt' changed type from 'CommerceDateTime!' to 'String!'
- :warning: Field 'CommerceOffer.submittedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOfferOrder.createdAt' changed type from 'CommerceDateTime!' to 'String!'
- :warning: Field 'CommerceOfferOrder.lastApprovedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOfferOrder.lastSubmittedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOfferOrder.stateExpiresAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOfferOrder.stateUpdatedAt' changed type from 'CommerceDateTime' to 'String'
- :warning: Field 'CommerceOfferOrder.updatedAt' changed type from 'CommerceDateTime!' to 'String!'
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8dc359030bb4261c1441ea295e16295b470fa4d5
</p>

File: src/desktop/assets/auction_reaction.js [hunk 1]
```diff
@@ -0,0 +1 @@
+import "desktop/apps/auction_reaction/client"
```
Comment (damassi) [inline src/desktop/assets/auction_reaction.js]: .js -> .ts

**Comment by dblandin** [comment]
@damassi Looks good to me, but would love another :mag_right: before merging.

**Comment by damassi** [comment]
Looks good!

File: src/desktop/index.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ app.use(
 app.use(require("./apps/home"))
 app.use(require("./apps/editorial_features"))
 app.use(require("./apps/apply"))
+app.use(require("./apps/auction_reaction/server").app)
```
Comment (dleve123) [inline src/desktop/index.js]: Do we prefer `server` over `index` now? Or is requiring a server file for different use-cases?

File: src/desktop/apps/auction_reaction/routes.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { buildServerApp } from "reaction/Artsy/Router/server"
+import { buildServerAppContext } from "desktop/lib/buildServerAppContext"
+import { routes } from "reaction/Apps/Auction/routes"
+import { stitch } from "@artsy/stitch"
+
+export const bidderRegistration = async (req, res, next) => {
+  if (!res.locals.sd.CURRENT_USER) {
```
Comment (dleve123) [inline src/desktop/apps/auction_reaction/routes.tsx]: this is effectively a "route-level validation"? Don't allow a client to see the registration UI unless there's a logged in user? Would we extend this to render a modal when a user is already registered for the sale?

File: src/desktop/apps/auction_reaction/routes.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      scripts,
+      styleTags,
+    } = await buildServerApp({
+      routes,
+      url: req.url,
+      userAgent: req.header("User-Agent"),
+      context: buildServerAppContext(req, res),
+    })
+
+    if (redirect) {
```
Comment (dleve123) [inline src/desktop/apps/auction_reaction/routes.tsx]: What's the purpose of this code? If there is a redirect prop passed into the component, then redirect the component and don't render the registration form? What's the use-case for this?

File: src/desktop/apps/auction_reaction/routes.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      res.redirect(302, redirect.url)
+      return
+    }
+
+    // Render layout
```
Comment (dleve123) [inline src/desktop/apps/auction_reaction/routes.tsx]: total nitpick – but don't think this comment adds much...is there any particular complexity you are trying to call out?

File: src/desktop/apps/auction_reaction/templates/index.jade [hunk 6]
```diff
@@ -0,0 +1,8 @@
+extends ../../../components/main_layout/templates/redesign
+
+append locals
+  - assetPackage = 'auction_reaction'
+
+block body
+  #react-root
```
Comment (dleve123) [inline src/desktop/apps/auction_reaction/templates/index.jade]: Is this the view rendered by the [`layout`](https://github.com/artsy/force/pull/4323/files?file-filters%5B%5D=.jade&file-filters%5B%5D=.styl&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#diff-9d2c1171b2d3fecb640db98d60802905R33) call?

Is there some magic that automatically renders the Reaction-housed registration form app? How does that work?

**Pull request merged**